src/pages/view-test-result/view-test-result.ts

```diff
@@ -1,26 +1,122 @@
+import { of } from 'rxjs/observable/of';
+import { TestDetailsModel } from './components/test-details/test-details.model';
+import { inflateSync } from 'zlib';
+import { Subscription } from 'rxjs/Subscription';
+import { DateTime } from '../../shared/helpers/date-time';
+import { ExaminerDetailsModel } from './components/examiner-details/examiner-details.model';
 
 @IonicPage()
 @Component({
   selector: 'page-view-test-result',
   templateUrl: 'view-test-result.html',
 })
-export class ViewTestResultPage extends BasePageComponent {
+export class ViewTestResultPage extends BasePageComponent implements OnInit, OnDestroy {
 
   applicationReference: string = '';
 
+  testResult: StandardCarTestCATBSchema;
+  testDetails: TestDetailsModel;
+  examinerDetails: ExaminerDetailsModel;
+
+  isLoading: boolean;
+  loadingSpinner: Loading;
+  subscription: Subscription;
+
   constructor(
     public navController: NavController,
     public platform: Platform,
     public navParams: NavParams,
+    public loadingController: LoadingController,
     public authenticationProvider: AuthenticationProvider,
+    public searchProvider: SearchProvider,
+
   ) {
     super(platform, navController, authenticationProvider);
 
     this.applicationReference = navParams.get('applicationReference');
   }
 
+  ngOnInit(): void {
+    this.handleLoadingUI(true);
+
+    this.subscription = this.searchProvider
+      .getTestResult(this.applicationReference, this.authenticationProvider.getEmployeeId())
+      .pipe(
+        tap((data) => {
+          const buffer = Buffer.from(data, 'base64');
+          const decompressedData = inflateSync(buffer).toString();
+          this.testResult = JSON.parse(decompressedData) as StandardCarTestCATBSchema;
+          this.generateData();
+          this.handleLoadingUI(false);
+        }),
+        catchError((error) => {
+          console.log('ERROR', JSON.stringify(error));
+          return of();
+        }),
+      ).subscribe();
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  handleLoadingUI = (isLoading: boolean): void => {
+    this.isLoading = isLoading;
+
+    if (isLoading) {
+      this.loadingSpinner = this.loadingController.create({
+        dismissOnPageChange: true,
+        spinner: 'circles',
+      });
+      this.loadingSpinner.present();
+      return;
+    }
+    if (this.loadingSpinner) {
+      this.loadingSpinner.dismiss();
+      this.loadingSpinner = null;
+    }
+  }
+
+  generateData(): void {
+    if (!this.testResult) {
+      return;
+    }
+
+    this.generateTestDetails();
+    this.generateExaminerDetails();
+
+  }
+
+  generateTestDetails() : void {
+
+    const applicationReference: ApplicationReference =  this.testResult.journalData.applicationReference;
+    const applicationReferenceString: string =
+      `${applicationReference.applicationId}${applicationReference.bookingSequence}${applicationReference.checkDigit}`;
```

Comment:
Chris mentioned that there's some common issue around doing a simple concatenation to make an app ref like this. There's something in the new [mes-microservice-common](https://github.com/dvsa/mes-microservice-common). May well be that this is all figured out on the backend, but worth checking.

Reply:
Looking at the code in there it seems like he wants to keep the application reference as a number instead of a string. As all we want to do is display it I can't see an issue with doing the string concatenation 

@chrisnappin  - Can you advise if there is something we need to watch out for.

Link to app ref code in mes-microservices-common - https://github.com/dvsa/mes-microservice-common/blob/develop/src/domain/tars.ts

Reply:
Yes, the booking sequence needs to get padded to two digits with leading zeros if needed, which is what the new backend common function does. Feel free to copy and paste it into the front end repo.